internal/notifications/notifications.go

```diff
@@ -0,0 +1,107 @@
+package notifications
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/frain-dev/convoy"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/internal/email"
+	"github.com/frain-dev/convoy/queue"
+	"github.com/frain-dev/convoy/util"
+	log "github.com/sirupsen/logrus"
+)
+
+type NotificationType string
+
+const (
+	SlackNotificationType NotificationType = "slack"
+	EmailNotificationType NotificationType = "email"
+)
+
+type Notification struct {
+	// Defines the type of notification either slack or email.
+	NotificationType NotificationType `json:"notification_type,omitempty"`
+
+	// Email or Slack notification
+	Payload interface{} `json:"payload,omitempty"`
+}
+
+type SlackNotification struct {
+	WebhookURL string `json:"webhook_url,omitempty"`
+
+	Text string `json:"text,omitempty"`
+}
+
+// NOTIFICATIONS
+
+func SendEndpointNotification(ctx context.Context,
+	app *datastore.Application,
+	endpoint *datastore.Endpoint,
+	group *datastore.Group,
+	status datastore.SubscriptionStatus,
+	q queue.Queuer,
+	failure bool,
+) error {
+	var ns []*Notification
+
+	if !util.IsStringEmpty(app.SupportEmail) {
+		_ = append(ns, &Notification{NotificationType: EmailNotificationType})
+	}
+
+	if !util.IsStringEmpty(app.SlackWebhookURL) {
+		_ = append(ns, &Notification{NotificationType: SlackNotificationType})
+	}
```

Comment:
Don't we need to make the slice a pointer when adding items to it? Since we're not assigning anything to the array how are we sure that ns contains the newly added items.

Reply:
Nice! Thanks! Fixed.